recipes/gperf/all/conanfile.py

```diff
@@ -27,9 +28,8 @@ def build_requirements(self):
                 self.build_requires("msys2/20190524")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version], filename=self.name + "-" + self.version + ".tar.gz")
+        os.rename(glob.glob("gperf-*")[0], self._source_subfolder)
```

Comment:
I've never tried but I think the filename does not matter with strip_root

```suggestion
        tools.get(**self.conan_data["sources"][self.version],
                  destination=self._source_subfolder, strip_root=True)
```

Reply:
@prince-chrismc Seems like that doesn't work with the `savannah` url we have for the source:
```
 tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
        ConanException: Cannot deduce file name from the url: 'https://git.savannah.gnu.org/gitweb/?p=gperf.git;a=snapshot;h=96ebd0b95dda2817923ef6de25cdaca1e7ee2956;sf=tgz'. Use 'filename' parameter.
```

Reply:
This worked for me.
```suggestion
        tools.get(**self.conan_data["sources"][self.version],
                  filename="gperf.tar.gz",
                  destination=self._source_subfolder, strip_root=True)
```

The build fails later because of missing `makeinfo` and `texi2pdf`.